plugins/scale/scale-title-overlay.cpp

```diff
@@ -0,0 +1,309 @@
+#include "scale-title-overlay.hpp"
+
+#include <wayfire/opengl.hpp>
+#include <wayfire/util/log.hpp>
+#include <wayfire/plugins/common/cairo-util.hpp>
+#include <wayfire/plugins/common/simple-texture.hpp>
+
+/**
+ * Class storing an overlay with a view's title, added to scale's transformer.
+ */
+class view_title_overlay_t
+{
+  protected:
+    wf::cairo_text_t overlay;
+    wf::cairo_text_t::params par;
+    /* the transformer we are attached to */
+    wf::scale_transformer_t& tr;
+    /* if text potentially overflows the available space */
+    bool overflow = false;
+
+    /**
+     * Get the transformed bounding box of the view, including the
+     * current transform
+     */
+    wlr_box get_transformed_bounding_box()
+    {
+        wlr_box box = tr.get_transformed_view()->get_wm_geometry();
```

Comment:
You should use the `get_bounding_box(nonstd::observer_ptr<transformer>)` functions, they will give you the bounding box up to the transformer, and then you can apply the last step manually. `wm_geometry` is rarely a good option since you can have scaling, etc.

Reply:
The problem is that `get_bounding_box()` includes shadows. See the attached screenshot where the gedit window was maximized, so no shadows, while the other ones were not, so there is an extra gap below the overlay. This is why I found it is better to use `get_wm_geometry()` and transform it ourselves: the `trasform_box_without_padding()` function that calls `transform_region(this)` to do the transform up to our transformer.

Reply:
![scale_overlay_padding](https://user-images.githubusercontent.com/6224679/104732647-8096f180-5778-11eb-940c-37c7f4cb2c53.png)

Reply:
I see, makes sense. Can we then rename the function to something that indicates it is not the full bounding box?